lib/datasource/index.ts

```diff
@@ -55,25 +59,139 @@ const datasources: Record<string, Datasource> = {
   terraform,
 };
 
-const cacheNamespace = 'datasource-releases';
+export function baseUrlLegacyMassager(sourceUrl: string): string {
+  let url: string = sourceUrl.trim();
 
-async function fetchReleases(
+  if (url.startsWith('git@')) {
+    url = 'ssh://' + url; // workaround for https://github.com/npm/hosted-git-info/issues/56
+  }
+  const parsedUrl = URL.parse(url);
+  const extraBaseUrls = [];
+  const getHostsFromRulesGithub = hostRules.hosts({ hostType: 'github' }) || [];
+  // istanbul ignore if
+  if (
+    getHostsFromRulesGithub.includes(parsedUrl.hostname) ||
+    (parsedUrl.hostname &&
+      (parsedUrl.hostname === 'github.com' ||
+        parsedUrl.hostname === 'www.github.com'))
+  ) {
+    // istanbul ignore if
+    if (url.startsWith('git:github.com/')) {
+      // github-url-from-git does not process git: without the forward slashes.Neither the help pages for git pull/push nor the git extended documentation on protocols specify that as a valid url. Was this related to some package-specific logic, e.g. npm returns the git url without the initial slashes after the protocol?
+      url = 'https://' + url.substr(4);
+    }
+    // Massage www out of github URL
+    url = url.replace('www.github.com', 'github.com');
```

Comment:
I think we shoud move the url replaces to be the first actions, like
* `www.github.com` => `github.com`
* `http://github.com` => ` https://github.com`
* and even more ... 


So it would simplify other conditionals